core/src/main/java/org/apache/calcite/sql/dialect/SnowflakeSqlDialect.java

```diff
@@ -36,6 +39,30 @@ public SnowflakeSqlDialect(Context context) {
     super(context);
   }
 
+  @Override public void unparseCall(final SqlWriter writer, final SqlCall call, final int leftPrec,
+      final int rightPrec) {
+    switch(call.getKind()) {
+    case ENDS_WITH:
+    case STARTS_WITH:
+      unparseEndsStartsWith(writer, call);
+      break;
+    default:
+      super.unparseCall(writer, call, leftPrec, rightPrec);
+    }
+  }
+
+	/** 
+	 * Remove underscore for {@code STARTS_WITH} and {@code ENDS_WITH} operators to 
+	 * comply with Snowflake syntax.
+	 */
+  private static void unparseEndsStartsWith(final SqlWriter writer, final SqlCall call) {
+    final String name = call.getKind() == SqlKind.ENDS_WITH ? "ENDSWITH" : "STARTSWITH";
```

Comment:
Is it convention here to not have underscore?

Reply:
That's just how Snowflake names those functions see: https://docs.snowflake.com/en/sql-reference/functions/startswith